tkg/managementcomponents/management_component_install.go

```diff
@@ -290,14 +290,18 @@ func InstallManagementComponents(mcip *ManagementComponentsInstallOptions) error
 
 // InstallKappController installs kapp-controller to the cluster
 func InstallKappController(clusterClient clusterclient.Client, kappControllerOptions KappControllerOptions) error {
+	// Ensure last-applied annotation is present before kapp-controller upgrade
+	if err := clusterClient.WaitForDeployment(constants.KappControllerDeploymentName, kappControllerOptions.KappControllerInstallNamespace); err == nil { // only attempt to add annotation when kapp-controller exists on the cluster
```

Comment:
**Question**: Do we want to use WaitForDeployment or just check if Deployment is there. WaitForDeployment actually makes sure the replicas are available. Do we need to check that here ?

Reply:
Hi Shyaam, thanks for pointing this out. Yeah we only want to use it to check if Deployment is there. I just reuse this function to differentiate the install/upgrade case

Reply:
I use `GetResource` instead to check existence of `kapp-controller`